fgpyo/util/tests/test_inspect.py

```diff
@@ -66,3 +74,23 @@ def test_attr_from_custom_type_without_parser_fails() -> None:
             kwargs={"foo": ""},
             parsers={},
         )
+
+
+def test_list_parser() -> None:
+    parser = list_parser(Foo, List[int], {})
+    assert parser("1,2,3") == [1, 2, 3]
+
+
+def test_set_parser() -> None:
+    parser = set_parser(Foo, Set[int], {})
+    assert parser("{1,2,3}") == {1, 2, 3}
+
+
+def test_tuple_parser() -> None:
+    parser = tuple_parser(Foo, Tuple[int, str], {})
+    assert parser("(1,a)") == (1, "a")
+
+
+def test_dict_parser() -> None:
+    parser = dict_parser(Foo, Dict[int, str], {})
+    assert parser("{123;a}") == {123: "a"}
```

Comment:
Could also check to parse `{}`, and `{123: "a", 123: "b"}`

Reply:
Here for the second case I did add an explicit check for duplicate keys